### ISSUE TITLE
[Bug]: False-Positive on a Page field with CaptionClass

### ISSUE THREAD

**Issue by vkozlov-sit** [ALCops/Analyzers#375]
### Analyzer

ApplicationCop

### Issue type

False positive (diagnostic fires when it shouldn't)

### Rule ID

AC0029

### Description

Hi

We have number of pages where field expression is defined by a field of a global record, and `CaptionClass` property points to its `FieldCaption` property. This is valid usage.

```al
field(FieldName; SomeRecord.SomeField)
{
    CaptionClass = SomeRecord.FieldCaption(SomeField);
    ToolTip = 'Specifies the Some Field of the Some Record.';
}
```

With that, we have `AC0029` warning as false-positive.
ToolTip is the same, in fact, as the one on the source table. But platform does not pick up the tooltip from the used Record variable, so it must be added.

**Expected**: not have any warnings for such page fields.

### AL code to reproduce

```al
field(CustomerNo; Customer."No.")
{
    CaptionClass = Customer.FieldCaption("No.");
    ToolTip = 'Specifies the number of the customer. The field is either filled automatically from a defined number series, or you enter the number manually because you have enabled manual number entry in the number-series setup.';
}

```

**Comment by Arthurvdv** [comment]
...or did you perhaps find a bug in the platform?

Currently the [AC0029](https://alcops.dev/docs/analyzers/applicationcop/ac0029/) is aligned with Microsoft's [AA0218](https://learn.microsoft.com/en-us/dynamics365/business-central/dev-itpro/developer/analyzers/codecop-aa0218) diagnostic. When I assign a `ToolTip` on the table, the AA0218 warning is suppressed for both fields, which seems to imply that the platform expects the table `ToolTip` to be inherited by the fields. I would otherwise expect the AA0218 to be raised on the `this.MyTable.MyField` field.

My suggestion would be to start by creating an issue in the Microsoft AL repository: https://github.com/Microsoft/AL. 

Either:
* You've found a bug in the platform, or
* You've found a bug in the CodeCop Analyzer rule AA0218

Depending on the outcome, we'll know whether AC0029 needs to be updated or not.

Would you like to do the honors of creating an issue in the official Microsoft repository?

<img width="1043" height="394" alt="Image" src="https://github.com/user-attachments/assets/985f388f-cb3e-40e9-8cbe-a252bcd6e665" />

```AL

page 50100 MyPage
{
    ApplicationArea = All;
    PageType = Card;
    SourceTable = MyTable;
    UsageCategory = Administration;

    layout
    {
        area(Content)
        {
            group(Group)
            {
                field(MyField; Rec.MyField)
                {
                    Caption = 'My Field (Rec.)';
                }
                field(MyFieldGlobalVar; this.MyTable.MyField)
                {
                    Caption = 'My Field (this.MyTable)';
                }
            }
        }
    }

    var
        MyTable: Record MyTable;

}
table 50100 MyTable
{
    fields
    {
        field(1; MyField; Integer)
        {
            Caption = 'My Field';
            ToolTip = 'Lorem ipsum dolor sit amet amet ut no magna eos sea sit nisl.';
        }
    }
}
```

**Comment by vkozlov-sit** [comment]
Thanks for checking, I'll send this to MS next week.
